### PR TITLE
Improve cross-module class isolation

### PR DESCRIPTION
**Pull request by Sanne**
https://issues.jboss.org/browse/ISPN-5410

### CODE REVIEW

File: as-modules/embedded/src/main/resources/org/infinispan/core/main/module.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <dependencies>
+        <module name="javax.api"/>
+        <module name="javax.transaction.api"/>
+        <module name="javax.xml.bind.api"/>
+        <module name="net.jcip"/>
+        <module name="org.apache.xerces" services="import"/>
+        <module name="org.infinispan.commons" slot="${slot}" />
+        <module name="org.infinispan.query" slot="${slot}" services="import" />
+        <module name="org.infinispan.lucene-directory" slot="${slot}" services="import" />
+        <module name="org.jboss.jandex"/>
```
Comment (tristantarrant) [inline as-modules/embedded/src/main/resources/org/infinispan/core/main/module.xml]: I think jandex can be killed

File: as-modules/embedded/src/main/resources/org/infinispan/core/main/module.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    </properties>
+    <resources>
+        <!-- Insert resources here -->
+    </resources>
+
+    <dependencies>
+        <module name="javax.api"/>
+        <module name="javax.transaction.api"/>
+        <module name="javax.xml.bind.api"/>
+        <module name="net.jcip"/>
```
Comment (tristantarrant) [inline as-modules/embedded/src/main/resources/org/infinispan/core/main/module.xml]: I think the JCIP annotations don't need to be kept around.

**Comment by Sanne** [comment]
Pushed a new version to also remove JCIP and Jandex as suggested.
That's 3 separate issues now:
 - https://issues.jboss.org/browse/ISPN-5410
 - https://issues.jboss.org/browse/ISPN-5418
 - https://issues.jboss.org/browse/ISPN-5419

File: as-modules/embedded/src/main/resources/org/infinispan/core/main/module.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        <property name="jboss.api" value="private"/>
+    </properties>
+    <resources>
+        <!-- Insert resources here -->
+    </resources>
+
+    <dependencies>
+        <module name="javax.api"/>
+        <module name="javax.transaction.api"/>
+        <module name="javax.xml.bind.api"/>
```
Comment (tristantarrant) [inline as-modules/embedded/src/main/resources/org/infinispan/core/main/module.xml]: I don't think we need JAXB anymore...

**Comment by tristantarrant** [comment]
Pushed, thanks

**Pull request closed without merging**